x-pack/legacy/plugins/ml/public/timeseriesexplorer/timeseriesexplorer.js

```diff
@@ -0,0 +1,1229 @@
+          mlJobSelectService.next({ selection: [jobs[0].id], resetSelection: true });
+        } else {
+        // Jobs exist, but no time series jobs.
+          this.setState({ loading: false });
+        }
+      }
+    } else {
+      this.setState({ loading: false });
+    }
+
+    // Reload the anomalies table if the Interval or Threshold controls are changed.
+    const tableControlsListener = () => {
+      const { zoomFrom, zoomTo } = this.state;
+      if (zoomFrom !== undefined && zoomTo !== undefined) {
+        this.loadAnomaliesTableData(zoomFrom.getTime(), zoomTo.getTime());
+      }
+    };
+
+    this.subscriptions.add(annotationsRefresh$.subscribe(this.refresh));
+    this.subscriptions.add(interval$.subscribe(tableControlsListener));
+    this.subscriptions.add(severity$.subscribe(tableControlsListener));
+    this.subscriptions.add(mlTimefilterRefresh$.subscribe(this.refresh));
+
+    // Listen for changes to job selection.
+    this.subscriptions.add(mlJobSelectService.subscribe(({ selection }) => {
+      const { appState } = this.props;
+      const { jobs } = this.state;
+
+      // Clear the detectorIndex, entities and forecast info.
+      if (selection.length > 0 && appState !== undefined) {
+        delete appState.mlTimeSeriesExplorer.detectorIndex;
+        delete appState.mlTimeSeriesExplorer.entities;
+        delete appState.mlTimeSeriesExplorer.forecastId;
+        appState.save();
+
+        this.setState({ showForecastCheckbox: false });
+        this.loadForJobId(selection[0], jobs);
+      }
+    }));
+
+    timefilter.enableTimeRangeSelector();
+    timefilter.enableAutoRefreshSelector();
+    timefilter.on('timeUpdate', this.refresh);
+
+    // Required to redraw the time series chart when the container is resized.
+    this.resizeChecker = new ResizeChecker(this.resizeRef.current);
+    this.resizeChecker.on('resize', () => {
+      this.resizeHandler();
+    });
+    this.resizeHandler();
+  }
+
+  componentWillUnmount() {
+    this.subscriptions.unsubscribe();
+    this.props.timefilter.off('timeUpdate', this.refresh);
+    this.resizeChecker.destroy();
+  }
+
+  render() {
+    const {
+      dateFormatTz,
+      globalState,
+      mlJobSelectService,
+      timefilter,
+    } = this.props;
+
+    const {
+      autoZoomDuration,
+      chartDetails,
+      contextAggregationInterval,
+      contextChartData,
+      contextForecastData,
+      dataNotChartable,
+      detectors,
+      detectorId,
+      entities,
+      focusAggregationInterval,
+      focusAnnotationData,
+      focusChartData,
+      focusForecastData,
+      hasResults,
+      jobs,
+      loading,
+      modelPlotEnabled,
+      selectedJob,
+      showAnnotations,
+      showAnnotationsCheckbox,
+      showForecast,
+      showForecastCheckbox,
+      showModelBounds,
+      showModelBoundsCheckbox,
+      svgWidth,
+      swimlaneData,
+      tableData,
+      zoomFrom,
+      zoomTo,
+    } = this.state;
+
+    const tsc = {
```

Comment:
As above, this needs a clearer name I think.

Reply:
Renamed to `chartProps` in 18a5945be7.